src/main/scala/optics/poly/Focus.scala

```diff
@@ -14,92 +13,29 @@ object dsl {
 }
 
-object GenLens {
-
-  def impl[From: Type, To: Type](getterExpr: Expr[From => To])(using quotes: Quotes): Expr[Lens[From, To]] = {
-    import quotes.reflect._
-
-    def unwrap(term: Term): Term = {
-      term match {
-        case Block(List(), inner) => unwrap(inner)
-        case Inlined(_, _, inner) => unwrap(inner)
-        case x => x
-      }
-    }
-
-    object ExpectedLambdaFunction {
-      def unapply(term: Term): Option[String] = unwrap(term) match {
-        case Lambda(List(ValDef(argName, _, _)), Select(Ident(identifier), fieldName)) if argName == identifier => Some(fieldName)
-        case _ => None
-      }
-    }
-
-    def constructGetterBody(field: String, from: Term): Term = {
-      // o.copy(field = value)
-      Select.unique(from, field)
-    }
-
-    def constructSetterBody(field: String, from: Term, to: Term): Term = {
-      // o.copy(field = value)
-      Select.overloaded(from, "copy", Nil, NamedArg(field, to) :: Nil)
-    }
-
-    def constructLensExpr(field: String): Expr[Lens[From, To]] = 
-      '{
-        val setter = (to: To) => (from: From) => ${
-          constructSetterBody(field, '{from}.asTerm, '{to}.asTerm).asExprOf[From]
-        }
-        
-        val getter = (from: From) => ${ 
-          constructGetterBody(field, '{from}.asTerm).asExprOf[To] 
-        }
-        Lens.apply(getter, setter)
-      }
-
-    getterExpr.asTerm match {
-      case ExpectedLambdaFunction(fieldName) =>
-        TypeRepr.of[From].classSymbol match {
-          case Some(fromClass) if fromClass.flags.is(Flags.Case) => constructLensExpr(fieldName)
-          case Some(fromClass) => report.error(s"Expecting a case class in the 'From' position; found $fromClass"); '{???}
-          case None => report.error(s"Expecting a concrete case class in the 'From' position; cannot reify type ${summon[Type[From]]}"); '{???}
-        }
-      case term =>
-        report.error(s"Expecting a lambda function that directly accesses a field. Example: `GenLens[Address](_.streetNumber)`")
-        '{???}
-    }
-  }
-
-  def apply[From] = new MkGenLens[From]
-
-  inline def uncurried[From, To](inline get: (From => To)): Lens[From, To] = 
-    ${ GenLens.impl('get) }
-
-  class MkGenLens[From] {
-    inline def apply[To](inline get: (From => To)): Lens[From, To] = 
-      ${ GenLens.impl('get) }
-  }
-
-}
-
 object Focus {
 
-  def impl[From: Type, To: Type](getterExpr: Expr[From => To])(using quotes: Quotes): Expr[Lens[From, To]] = {
+  def impl[From: Type, To: Type](getterExpr: Expr[From => To])(using Quotes): Expr[Lens[From, To]] = {
     import quotes.reflect._
-  
 
     ///////////// DOMAIN //////////////////
     case class ParseParams(argName: String, argType: TypeRepr, lambdaBody: Term)
 
+    // Common type information that we record about every action in the DSL
+    case class TypeInfo(from: TypeRepr, fromTypeArgs: List[TypeRepr], to: TypeRepr) {
```

Comment:
Could you show me an example of `TypeInfo` parameters for a particular field? I am not familiar with `TypeRepr`, I am not sure to see how it would look like.

Reply:
Nice, thanks for the example.